src/semidiscretization/semidiscretization.jl

```diff
@@ -49,7 +49,8 @@ function integrate(func::Func, u_ode, semi::AbstractSemidiscretization;
 end
 
 function integrate(u, semi::AbstractSemidiscretization; normalize = true)
-    integrate(cons2cons, u, semi; normalize = normalize)
+    tmp_u = copyto!(CPU(), allocate(CPU(), eltype(u), size(u)), u)
+    integrate(cons2cons, tmp_u, semi; normalize = normalize)
```

Comment:
Just curious: is it not possible (or feasible) to execute integration on the GPU, or is it just not implemented yet?

Reply:
Another question: if the array `u` already lives on the CPU, is this still a copy (I assume it is) or is it a no-op. If it forces a copy, we should consider dispatching on `u`, i.e., if it is our "CPU backend array type" we should keep the original implementation and only do the copy on non-CPU backends.

But this is really just something to keep in mind/store on a TODO list, not something that needs to be done right now